examples/01_encodings.py

```diff
@@ -61,75 +61,110 @@
+# We can do this easily using skrub's |TableVectorizer|
 
-from sklearn.ensemble import HistGradientBoostingRegressor
-from sklearn.pipeline import make_pipeline
 from skrub import TableVectorizer
 
-pipeline = make_pipeline(TableVectorizer(), HistGradientBoostingRegressor())
-pipeline.fit(X, y)
+vectorizer = TableVectorizer()
+features = vectorizer.fit_transform(employees)
+features
 
 ###############################################################################
-# What just happened here?
-#
-# We actually gave our dataframe as an input to the |TableVectorizer| and it
-# returned an output useful for the scikit-learn model.
-#
-# Let's explore the internals of our encoder, the |TableVectorizer|:
+# From our 8 columns, the |TableVectorizer| has extracted 143 numerical
+# features. Most of them are one-hot encoded representations of the categorical
+# features. For example, we can see that 3 columns ``gender_F``, ``gender_M``,
+# ``gender_nan`` were created to encode the ``'gender'`` column.
 
-from pprint import pprint
+###############################################################################
+# Before we explore the parameters of the |TableVectorizer| and the choices it
+# made, let us note that by performing apropriate transformations on our
+# complex data, it allows us to use our table for machine-learning:
 
-# Recover the TableVectorizer from the Pipeline
-tv = pipeline.named_steps["tablevectorizer"]
+from sklearn.ensemble import HistGradientBoostingRegressor
 
-pprint(tv.transformers_)
+HistGradientBoostingRegressor().fit(features, salaries)
 
 ###############################################################################
-# We observe it has automatically assigned an appropriate encoder to
-# corresponding columns:
+# The |TableVectorizer| bridges the gap between tabular data and machine-learning
+# pipelines and allows us to apply a machine-learning estimator to our table
+# without manual data wrangling and feature extraction.
+#
 
 ###############################################################################
-#     - The |OneHotEncoder| for low cardinality string variables, the columns
-#       ``'gender'``, ``'department'``, ``'department_name'`` and ``'assignment_category'``.
+# Inspecting the TableVectorizer
+# ------------------------------
+#
+# The |TableVectorizer| distinguishes between 4 basic types of columns.
+# For each kind, it applies a different transformation, which we can configure.
+# The types of columns and the default transformations for each of them are:
+#
+# - numeric columns: simply casting to float32
+# - datetime columns: extracting features such as year, day, hour with the |DatetimeEncoder|
+# - low-cardinality categorical columns: one-hot encoding
+# - high-cardinality categorical columns: a simple and effective text representation pipeline provided by the |GapEncoder|
+#
+
+vectorizer
+
+# We can inspect which transformation was chosen for a each column and retrieve the fitted transformer.
```

Comment:
for narrative text I kind of like having each sentence on one line because it tends to result in more meaningful diffs than re-wrapping paragraphs with every change. WDYT?

Reply:
you're right, I was looking at an older version which was a commentary in a code box, and the commentary was too long